get-build-times/get-build-times.ts

```diff
@@ -0,0 +1,125 @@
+import { createObjectCsvWriter } from "csv-writer";
+import {
+  AmplifyClient,
+  ListJobsCommand,
+  ListJobsCommandInput,
+  ListAppsCommandInput,
+  ListAppsCommand,
+  App,
+  JobStatus,
+  ListJobsCommandOutput,
+} from "@aws-sdk/client-amplify";
+
+const amplifyClient = new AmplifyClient({
+  region: "ap-southeast-1",
+});
+
+const branchName = "staging";
+
+// Define the path and name of the CSV file to write to
+const csvFilePath = "build-times.csv";
+
+// Define the CSV header row
+const csvHeader = [
+  { id: "timeToLive", title: "Time To Live" },
+  { id: "buildTime", title: "Build Time" },
+  { id: "name", title: "Name" },
+];
+
+// Define the CSV writer
+const csvWriter = createObjectCsvWriter({
+  path: csvFilePath,
+  header: csvHeader,
+});
+
+const createListJobsCommand = (input: ListJobsCommandInput) => {
+  return new ListJobsCommand(input);
+};
+
+const createListAppsCommand = (input: ListAppsCommandInput) => {
+  return new ListAppsCommand(input);
+};
+
+// get all app with pagination
+const getAllApps = async () => {
+  let token: string | undefined;
+  const apps: App[] = [];
+  do {
+    const listAppsCommand = createListAppsCommand({
+      nextToken: token,
```

Comment:
my assumption here is that on initial load, this is `undefined` and hence signifies to amplify that this is the initial query?

thereafter, we submit the token we get back from amplify?

Reply:
yup!